### PR TITLE
MM-32018 update URL endpoint for DownloadComplianceReport function

### PR DESCRIPTION
**Pull request by maxerenberg**
#### Summary
This pull request fixes the DownloadComplianceReport method in the golang client so that a full report is downloaded, not just the metadata.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32018

#### Release Note
```release-note
The DownloadComplianceReport function in the golang driver has been fixed to download
a full report as a zip archive.
```

### CODE REVIEW

File: model/client4.go [hunk 1]
```diff
@@ -386,6 +386,10 @@ func (c *Client4) GetComplianceReportRoute(reportId string) string {
 	return fmt.Sprintf("/compliance/reports/%v", reportId)
 }
 
+func (c *Client4) GetComplianceReportDownloadRoute(reportId string) string {
+	return fmt.Sprintf("/compliance/reports/%v/download", reportId)
+}
```
Comment (sbishel) [inline model/client4.go:391]: I see you followed the pattern used in the above function, but let's update both `GetComplianceReportDownloadRoute` and `GetComplianceReportRoute` to use `GetComplianceReportsRoute`.

**Review by sbishel — CHANGES_REQUESTED** [review]
Nice find....

**Review by sbishel — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Test cases passed
Golang client working as expected

image

**Comment by amyblais** [comment]
@maxerenberg @mkraft It seems this missed a second dev review - please see the process here https://developers.mattermost.com/contribute/getting-started/code-review/.

**Comment by maxerenberg** [comment]
My apologies. I'll make sure to check the Code Review process for next time.

**Pull request merged**